app/src/main/java/com/example/drawingApp/DialogObject.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.drawingApp
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogObject {
+
+    fun showLayerAlertDialog(
+        c: Context,
+        model: DrawingViewModel,
+        onDialogSubmission: (String) -> Unit
+    ) {
+        val inflater = LayoutInflater.from(c)
+        val view = inflater.inflate(R.layout.layer_dialog, null)
+        val layerField = view.findViewById<EditText>(R.id.newLayerField)
+        val errorText = view.findViewById<TextView>(R.id.dialogError)
+        val layerNames = mutableListOf<String>()
+        for (x in model.layerNames) layerNames.add(x.text)
+        var layerCounter = 1
+
+
+        //find the first available layer name not taken
+        while (layerNames.contains(
+                c.resources.getString(
+                    R.string.new_layer_hint,
+                    layerCounter
+                )
+            )
+        ) {
+            layerCounter += 1
+        }
+        layerField.setText(c.resources.getString(R.string.new_layer_hint, layerCounter))
+
+        //needs to be after the set text or else the textchanged listener will overwrite the previous text needed for rotations
+        layerField.addTextChangedListener {
+            val fieldString = it.toString().trim()
```

Comment:
unsure why we used .trim() instead of just working with the string, Logic should flexible to handle a string rather than needing it trimmed

Reply:
because the trim is what we are actually going to be taking for the value, if its not valid in that form the user needs to know, no reason to do the hard work of rewriting a trim for it. The trim function really just sends back the string with no whitespace in front or behind it.